source/obstacle/DeadlyObstacle.hx

```diff
@@ -1,19 +1,19 @@
 package obstacle;
 
+import flixel.FlxG;
 import flixel.FlxBasic;
 import flixel.util.FlxColor;
 import obstacle.Obstacle.ObstacleParameters;
 
-class DeadlyObstacle extends Obstacle
-{
-	override function init(parameters:ObstacleParameters)
-	{
+class DeadlyObstacle extends Obstacle {
+	override function init(parameters:ObstacleParameters) {
 		super.init(parameters);
 		color = FlxColor.RED;
+		velocity.y = 0;
+		this.y = FlxG.random.float(this.height, FlxG.height - this.height);
```

Comment:
Rather than force this kind of randomization in the `init()` method, you get more flexibility if the deadly obstacle generator has the appropriate parameters / variation to get the kind of range you want.

If you specifically wanted to make sure that the obstacles wouldn't end up fully or partially off-screen, I think you could still use params + variation and do extra logic here to help them get on-screen if they ended up partially off.